### PR TITLE
GN-4748: add support for the creation of decision templates

### PR DESCRIPTION
**Pull request by elpoelma**
## Overview
This PR introduces support for the creation of decision templates. 
Specifically it includes the following changes:
- A rename of the `regulatory-attachments` route to `templates-management`
- The introduction and removal of the following models based on the new data model
  * [+] `template`
  * [+] `template-version`
  * [+] `regulatory-attachment-template`
  * [+] `decision-template`
  * [-] `published-regulatory-attachment`
  * [-] `published-regulatory-attachment-container`
- Modification of `templates-management` routes to correctly support the creation/editing/publication of both regulatory attachment and decision templates

#### connected issues and PRs:
[GN-4748](https://binnenland.atlassian.net/browse/GN-4748?atlOrigin=eyJpIjoiYzhmYTJiY2NhYmRlNDdhZDhmMzk1MTM0OWEyMjQ3ZTciLCJwIjoiaiJ9)
https://github.com/lblod/app-reglementaire-bijlage/pull/77
https://github.com/lblod/reglement-publish-service/pull/14
https://github.com/lblod/frontend-gelinkt-notuleren/pull/635

plugins PR: https://github.com/lblod/ember-rdfa-editor-lblod-plugins/pull/440
editor PR: https://github.com/lblod/ember-rdfa-editor/pull/1197
Solves https://github.com/lblod/frontend-reglementaire-bijlage/issues/227

### TODO
- [x] Fix template removal
- [x] Internationalization
- [x] Indicate template type on `publish` route

### Setup
Check-out https://github.com/lblod/app-reglementaire-bijlage/pull/77 for a description on how to set-up the stack

for the frontend, just npm install and run, the correct dev builds of the plugins and editor are set up

### How to test/reproduce
Check-out https://github.com/lblod/app-reglementaire-bijlage/pull/77 for a description on how to test the new features.

### Checks PR readiness
- [x] UI: works on smaller screen sizes
- [x] UI: feedback for any loading/error states
- [x] Check cancel/go-back flows
- [x] Check database state correct when deleting/updating (especially regarding relationships)
- [x] changelog
- [x] npm lint

### CODE REVIEW

File: app/controllers/templates-management/index.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    this.editorDocument = undefined;
+    this.removeTemplateModalIsOpen = false;
+  }
+
+  submitRemoveTemplate = task(async () => {
+    // this is currenlty a hard-delete of the container, but a soft-delete of the
+    // publishedVersion, since GN filters on the validThrough date
+    // if we'd want to soft-delete the container as well
+    // we'd likely have to use RS_DELETED_FOLDER from utils/constants
+    console.log(this.documentContainer.id);
```
Comment (piemonkey) [inline app/controllers/templates-management/index.js:133]: Console.log

File: app/controllers/templates-management/publish.js [hunk 2]
```diff
@@ -19,39 +24,29 @@ export default class PublishController extends Controller {
...
+    const currentPublishedTemplate = (
+      await this.store.query('template', {
+        filter: {
+          'derived-from': {
+            id: this.model.container.id,
+          },
+        },
+      })
+    )[0];
+    console.log(currentPublishedTemplate);
```
Comment (piemonkey) [inline app/controllers/templates-management/publish.js:36]: Console.log

File: app/controllers/templates-management/publish.js [hunk 3]
```diff
@@ -19,39 +24,29 @@ export default class PublishController extends Controller {
...
+    if (currentPublishedTemplate) {
       const publishedVersion =
-        await publishedVersionContainer.currentVersion.reload();
+        await currentPublishedTemplate.currentVersion.reload();
```
Comment (piemonkey) [inline app/controllers/templates-management/publish.js:39]: Is it possible to do this in one go with the query for the template?

File: app/router.js [hunk 4]
```diff
@@ -14,7 +14,7 @@ Router.map(function () {
     this.route('cookiestatement');
     this.route('accessibilitystatement');
   });
-  this.route('regulatory-attachments', function () {
+  this.route('templates-management', function () {
```
Comment (piemonkey) [inline app/router.js:17]: I feel bad for saying this, but my brain can't let it go with all the related file names as well... I think it should be 'template-management' to be correct English...

**Review by piemonkey — APPROVED** [review]
Looks good to me. Only very pedantic comments.

File: app/router.js (hunk 4 above)
Comment (elpoelma) [reply app/router.js:17]: Yes agreed, we should then also rename `codelists-management` to `codelist-management` and `snippets-management` to `snippet-management`. I'll do the renames in a seperate PR.

File: app/controllers/templates-management/publish.js (hunk 3 above)
Comment (elpoelma) [reply app/controllers/templates-management/publish.js]: Good point! I added an `include: 'current-version'` to the template query.

File: app/router.js (hunk 4 above)
Comment (elpoelma) [reply app/router.js:17]: Should be adjusted in https://github.com/lblod/frontend-reglementaire-bijlage/pull/238 :)

**Review by abeforgit — CHANGES_REQUESTED** [review]
the decision templates use the wrong editor config, they need to use  the decision config like GN, so they can insert motivations, descriptions, and decision-articles (which are not (yet) compatible with structure-articles)

**Comment by piemonkey** [comment]
Should be all switched now. I hope I've not missed differences between the two, I'm not very familiar with exactly what differences there should be, I just based it off the differences in GN.

**Comment by abeforgit** [comment]
@elpoelma I can't request you as a reviewer cause you opened the PR, but would appreciate a look

**Comment by abeforgit** [comment]
@piemonkey @dkozickis this is finally ready for another look

**Review by piemonkey — APPROVED** [review]
I'm having trouble with mu-auth on my local app-rb, so haven't been able to test yet, but the changes since I touched this look good in code.

**Pull request merged**